public/pages/Detectors/components/UpdateAlertConditions/UpdateAlertConditions.tsx

```diff
@@ -74,8 +84,24 @@ export default class UpdateAlertConditions extends Component<
 
   getRules = async () => {
     try {
-      const { ruleService } = this.props;
+      const { ruleService, detectorService } = this.props;
       const { detector } = this.state;
+      if (!detector.name) {
+        const response = (await detectorService.getDetectors()) as ServerResponse<
+          SearchDetectorsResponse
+        >;
+        if (response.ok) {
+          const detectorHit = response.response.hits.hits.find(
+            (detectorHit) => detectorHit._id === detector.id
+          ) as DetectorHit;
```

Comment:
We should instead use the get detector API using the id - https://opensearch.org/docs/latest/security-analytics/api-tools/detector-api/#get-detector

Reply:
@amsiglan I noticed we are using this method in other places as well and as I can see, the reason behind it is that the get detector doesn't return the triggers array in its response, and we need this for alerts. Perhaps, we could add a feature request for BE to add triggers in response as well?

Reply:
aah, let me add an issue, but most likely it won't be available before 2.6, so we can go ahead with current approach for now